app/scripts/lib/offscreen-bridge/trezor-offscreen-bridge.ts

```diff
@@ -0,0 +1,142 @@
+      if (
+        msg.target === OffscreenCommunicationTarget.extension &&
+        msg.event === OffscreenCommunicationEvents.trezorDeviceConnect
+      ) {
+        this.model = msg.payload;
+      }
+    });
+
+    return new Promise<void>((resolve) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.init,
+          params: settings,
+        },
+        () => {
+          resolve();
+        },
+      );
+    });
+  }
+
+  dispose() {
+    return new Promise<void>((resolve) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.dispose,
+        },
+        () => {
+          resolve();
+        },
+      );
+    });
+  }
+
+  getPublicKey(params: { path: string; coin: string }) {
+    return new Promise((resolve) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.getPublicKey,
+          params,
+        },
+        (response) => {
+          resolve(
+            response as unknown as
+              | Unsuccessful
+              | Success<{ publicKey: string; chainCode: string }>,
+          );
+        },
+      );
+    }) as TrezorResponse<{ publicKey: string; chainCode: string }>;
+  }
+
+  ethereumSignTransaction(params: Params<EthereumSignTransaction>) {
+    return new Promise((resolve) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.signTransaction,
+          params,
+        },
+        (response) => {
+          resolve(
+            response as unknown as Unsuccessful | Success<EthereumSignedTx>,
+          );
+        },
+      );
+    }) as TrezorResponse<EthereumSignedTx>;
+  }
+
+  ethereumSignMessage(params: Params<EthereumSignMessage>) {
+    return new Promise((resolve) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.signMessage,
+          params,
+        },
+        (response) => {
+          resolve(
+            response as unknown as
+              | Unsuccessful
+              | Success<PROTO.MessageSignature>,
+          );
+        },
+      );
+    }) as TrezorResponse<PROTO.MessageSignature>;
+  }
+
+  ethereumSignTypedData<T extends EthereumSignTypedDataTypes>(
+    params: Params<EthereumSignTypedHash<T>>,
+  ) {
+    return new Promise((resolve) => {
+      chrome.runtime.sendMessage(
+        {
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.signTypedData,
+          params,
+        },
+        (response) => {
+          resolve(
+            response as unknown as
+              | Unsuccessful
+              | Success<PROTO.EthereumTypedDataSignature>,
+          );
```

Comment:
Now that I added types in global.d.ts i can get rid of this casting here I believe.